src/mongo/registerMongoCommands.ts

```diff
@@ -86,20 +86,21 @@ export function registerMongoCommands(context: vscode.ExtensionContext, actionHa
     actionHandler.registerCommand('cosmosDB.launchMongoShell', launchMongoShell);
     actionHandler.registerCommand('cosmosDB.newMongoScrapbook', async () => await vscodeUtil.showNewFile('', context.extensionPath, 'Scrapbook', '.mongo'));
     actionHandler.registerCommand('cosmosDB.executeMongoCommand', async function (this: IActionContext) {
-        if (!connectedDb) {
-            const persistedNodeId: string | undefined = context.globalState.get(connectedDBKey);
-            if (persistedNodeId) {
-                const persistedNode: IAzureNode | undefined = await tree.findNode(persistedNodeId);
-                if (persistedNode) {
-                    await vscode.commands.executeCommand('cosmosDB.connectMongoDB', persistedNode);
-                }
-            }
-        }
-
-        await MongoCommands.executeCommandFromActiveEditor(connectedDb, context.extensionPath, editorManager, tree, this);
+        await loadPersistedMongoDBTask;
+        await MongoCommands.executeCommandFromActiveEditor(<IAzureParentNode<MongoDatabaseTreeItem>>ext.connectedMongoDB, context.extensionPath, editorManager, tree, this);
     });
 }
 
+async function loadPersistedMongoDB(context: vscode.ExtensionContext, tree: AzureTreeDataProvider): Promise<void> {
+    const persistedNodeId: string | undefined = context.globalState.get(connectedDBKey);
+    if (persistedNodeId) {
+        const persistedNode: IAzureNode | undefined = await tree.findNode(persistedNodeId);
+        if (persistedNode) {
+            await vscode.commands.executeCommand('cosmosDB.connectMongoDB', persistedNode);
```

Comment:
What if this fails?

Reply:
It feels like worst case you just don't get the 'persisted' behavior and you have to reconnect? Can you think of anything worse? That doesn't seem too bad to me

Reply:
You're awaiting it in execute, so it will fail every time.

Reply:
Can expand on that? I don't know what you mean...are you confusing `cosmosDB.executeMongoCommand` or `MongoCommands.executeCommandFromActiveEditor` with `vscode.commands.executeCommand`

Reply:
Never mind - I get you

Reply:
Fixed